File: LeetCode Problems/2001 to 2100/2038. Remove Colored Pieces if Both Neighbors are the Same Color/2038. Remove Colored Pieces if Both Neighbors are the Same Color.java
class Solution {
    private String s;
    private boolean check(int a, int b) {
        return s.charAt(a) == s.charAt(b);
    }
    public boolean winnerOfGame(String colors) {
        int A = 0, B = 0;
        s = colors;
        for (int i = 1; i < colors.length() - 1; i++) {
            if (check(i, i - 1) && check(i, i + 1)) {                    
                if(colors.charAt(i) == 'A') 
                    A++;
                else 
                    B++;
            }
        }
        if(A > B) 
            return true;
        return false;
    }
}

File: LeetCode Problems/2001 to 2100/2038. Remove Colored Pieces if Both Neighbors are the Same Color/2038. Remove Colored Pieces if Both Neighbors are the Same Color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int A = 0, B = 0;
        for (int i = 1; i < colors.size() - 1; i++) {
            if (colors[i] == colors[i - 1] && colors[i] == colors[i + 1]) {
                if (colors[i] == 'A')
                    A++;
                else
                    B++;
            }
        }
        if (A > B)
            return true;
        return false;
    }
};